vault-core/src/main/java/org/apache/jackrabbit/vault/fs/impl/io/DocViewSAXImporter.java

```diff
@@ -1295,6 +1297,37 @@ public void endElement(String uri, String localName, String qName) throws SAXExc
         }
     }
 
+    private boolean hasSiblingWithSameType(Node child) throws RepositoryException {
+
+        Node parent = child.getParent();
+        EffectiveNodeType ent = EffectiveNodeType.ofNode(parent);
+        Optional<NodeDefinition> childDef = ent.getApplicableChildNodeDefinition(child.getName(), child.getPrimaryNodeType());
+
+        if (!childDef.isPresent()) {
+            log.debug("no NodeDefinition for {}", child.getPath());
+        } else {
+            String typeName = childDef.get().getDeclaringNodeType().getName();
```

Comment:
why is the typename relevant? it's the mandatory type that needs to be compared.